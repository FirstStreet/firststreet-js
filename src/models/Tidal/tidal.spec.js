const Tidal = require('./index');
const tidalMock = require('../../__mocks__/tidalProperty');

let data = null;

describe('Tidal', () => {
  beforeEach(() => {
    data = new Tidal(tidalMock);
  });

  it('should initialize', () => {
    expect(data).toMatchSnapshot();
  });

  it('should have maxDepth', () => {
    expect(data.maxDepth('em', 2023)).toBeDefined();
  });

  it('should have lot', () => {
    expect(data.lot('kt', 2018)).toBeDefined();
  });

  it('should have neighborhood', () => {
    expect(data.neighborhood('em', 2033)).toBeDefined();
  });

  it('should have road', () => {
    expect(data.road('kt', 2023)).toBeDefined();
  });
});
