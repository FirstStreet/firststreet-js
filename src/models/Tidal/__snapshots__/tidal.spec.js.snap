// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tidal should initialize 1`] = `
Tidal {
  "currentData": Array [
    Object {
      "floodData": Array [
        Object {
          "data": Array [
            Object {
              "type": "maxdepth",
              "unit": "meters",
              "value": 0,
            },
            Object {
              "type": "lot",
              "unit": "percent",
              "value": 0,
            },
            Object {
              "type": "neighborhood",
              "unit": "percent",
              "value": 0.09542,
            },
            Object {
              "type": "road",
              "unit": "percent",
              "value": 0.05723,
            },
          ],
          "year": 2018,
        },
        Object {
          "data": Array [
            Object {
              "type": "maxdepth",
              "unit": "meters",
              "value": 0,
            },
            Object {
              "type": "lot",
              "unit": "percent",
              "value": 0,
            },
            Object {
              "type": "neighborhood",
              "unit": "percent",
              "value": 0.09924,
            },
            Object {
              "type": "road",
              "unit": "percent",
              "value": 0.0619,
            },
          ],
          "year": 2023,
        },
        Object {
          "data": Array [
            Object {
              "type": "maxdepth",
              "unit": "meters",
              "value": 0,
            },
            Object {
              "type": "lot",
              "unit": "percent",
              "value": 0,
            },
            Object {
              "type": "neighborhood",
              "unit": "percent",
              "value": 0.12977,
            },
            Object {
              "type": "road",
              "unit": "percent",
              "value": 0.06892,
            },
          ],
          "year": 2028,
        },
        Object {
          "data": Array [
            Object {
              "type": "maxdepth",
              "unit": "meters",
              "value": 0,
            },
            Object {
              "type": "lot",
              "unit": "percent",
              "value": 0,
            },
            Object {
              "type": "neighborhood",
              "unit": "percent",
              "value": 0.14885,
            },
            Object {
              "type": "road",
              "unit": "percent",
              "value": 0.07514,
            },
          ],
          "year": 2033,
        },
      ],
      "floodID": "em",
    },
    Object {
      "floodData": Array [
        Object {
          "data": Array [
            Object {
              "type": "maxdepth",
              "unit": "meters",
              "value": 0,
            },
            Object {
              "type": "lot",
              "unit": "percent",
              "value": 0,
            },
            Object {
              "type": "neighborhood",
              "unit": "percent",
              "value": 0.05725,
            },
            Object {
              "type": "road",
              "unit": "percent",
              "value": 0.02527,
            },
          ],
          "year": 2018,
        },
        Object {
          "data": Array [
            Object {
              "type": "maxdepth",
              "unit": "meters",
              "value": 0,
            },
            Object {
              "type": "lot",
              "unit": "percent",
              "value": 0,
            },
            Object {
              "type": "neighborhood",
              "unit": "percent",
              "value": 0.05725,
            },
            Object {
              "type": "road",
              "unit": "percent",
              "value": 0.02921,
            },
          ],
          "year": 2023,
        },
        Object {
          "data": Array [
            Object {
              "type": "maxdepth",
              "unit": "meters",
              "value": 0,
            },
            Object {
              "type": "lot",
              "unit": "percent",
              "value": 0,
            },
            Object {
              "type": "neighborhood",
              "unit": "percent",
              "value": 0.0687,
            },
            Object {
              "type": "road",
              "unit": "percent",
              "value": 0.03647,
            },
          ],
          "year": 2028,
        },
        Object {
          "data": Array [
            Object {
              "type": "maxdepth",
              "unit": "meters",
              "value": 0,
            },
            Object {
              "type": "lot",
              "unit": "percent",
              "value": 0,
            },
            Object {
              "type": "neighborhood",
              "unit": "percent",
              "value": 0.08015,
            },
            Object {
              "type": "road",
              "unit": "percent",
              "value": 0.04438,
            },
          ],
          "year": 2033,
        },
      ],
      "floodID": "kt",
    },
  ],
  "data": Object {
    "FSID": 450350219571,
    "results": Array [
      Object {
        "floodData": Array [
          Object {
            "data": Array [
              Object {
                "type": "maxdepth",
                "unit": "meters",
                "value": 0,
              },
              Object {
                "type": "lot",
                "unit": "percent",
                "value": 0,
              },
              Object {
                "type": "neighborhood",
                "unit": "percent",
                "value": 0.09542,
              },
              Object {
                "type": "road",
                "unit": "percent",
                "value": 0.05723,
              },
            ],
            "year": 2018,
          },
          Object {
            "data": Array [
              Object {
                "type": "maxdepth",
                "unit": "meters",
                "value": 0,
              },
              Object {
                "type": "lot",
                "unit": "percent",
                "value": 0,
              },
              Object {
                "type": "neighborhood",
                "unit": "percent",
                "value": 0.09924,
              },
              Object {
                "type": "road",
                "unit": "percent",
                "value": 0.0619,
              },
            ],
            "year": 2023,
          },
          Object {
            "data": Array [
              Object {
                "type": "maxdepth",
                "unit": "meters",
                "value": 0,
              },
              Object {
                "type": "lot",
                "unit": "percent",
                "value": 0,
              },
              Object {
                "type": "neighborhood",
                "unit": "percent",
                "value": 0.12977,
              },
              Object {
                "type": "road",
                "unit": "percent",
                "value": 0.06892,
              },
            ],
            "year": 2028,
          },
          Object {
            "data": Array [
              Object {
                "type": "maxdepth",
                "unit": "meters",
                "value": 0,
              },
              Object {
                "type": "lot",
                "unit": "percent",
                "value": 0,
              },
              Object {
                "type": "neighborhood",
                "unit": "percent",
                "value": 0.14885,
              },
              Object {
                "type": "road",
                "unit": "percent",
                "value": 0.07514,
              },
            ],
            "year": 2033,
          },
        ],
        "floodID": "em",
      },
      Object {
        "floodData": Array [
          Object {
            "data": Array [
              Object {
                "type": "maxdepth",
                "unit": "meters",
                "value": 0,
              },
              Object {
                "type": "lot",
                "unit": "percent",
                "value": 0,
              },
              Object {
                "type": "neighborhood",
                "unit": "percent",
                "value": 0.05725,
              },
              Object {
                "type": "road",
                "unit": "percent",
                "value": 0.02527,
              },
            ],
            "year": 2018,
          },
          Object {
            "data": Array [
              Object {
                "type": "maxdepth",
                "unit": "meters",
                "value": 0,
              },
              Object {
                "type": "lot",
                "unit": "percent",
                "value": 0,
              },
              Object {
                "type": "neighborhood",
                "unit": "percent",
                "value": 0.05725,
              },
              Object {
                "type": "road",
                "unit": "percent",
                "value": 0.02921,
              },
            ],
            "year": 2023,
          },
          Object {
            "data": Array [
              Object {
                "type": "maxdepth",
                "unit": "meters",
                "value": 0,
              },
              Object {
                "type": "lot",
                "unit": "percent",
                "value": 0,
              },
              Object {
                "type": "neighborhood",
                "unit": "percent",
                "value": 0.0687,
              },
              Object {
                "type": "road",
                "unit": "percent",
                "value": 0.03647,
              },
            ],
            "year": 2028,
          },
          Object {
            "data": Array [
              Object {
                "type": "maxdepth",
                "unit": "meters",
                "value": 0,
              },
              Object {
                "type": "lot",
                "unit": "percent",
                "value": 0,
              },
              Object {
                "type": "neighborhood",
                "unit": "percent",
                "value": 0.08015,
              },
              Object {
                "type": "road",
                "unit": "percent",
                "value": 0.04438,
              },
            ],
            "year": 2033,
          },
        ],
        "floodID": "kt",
      },
    ],
  },
}
`;
